pkg/controller/controller.go

```diff
@@ -185,6 +189,17 @@ func (c *controller) Run(workers int, stopCh <-chan struct{}) {
 		}
 	}
 
+	c.clusterIDLock.Lock()
+	c.clusterID = string(uuid.NewUUID())
```

Comment:
I would rather see this delayed until a new cluster ID is needed. As it is now, there is a period of time at the start of the controller where the cluster ID held in the controller will be different than the cluster ID that may exist in the cluster-info config map. I would like the controller to fail reconciliation of instances until the cluster ID.

Reply:
made it lazy on first get. need a look to check it looks okay